### ISSUE TITLE
Block toolbar not showing on iOS

### ISSUE THREAD

**Issue by ColinRobbins** [WordPress/gutenberg#53495]
### Description

When editing a post / page the block editor is not showing in the tool bar at the top of the screen.
On further inspection, it is there *you can click items*, but they are not visible.

### Step-by-step reproduction instructions

Edit a page in Gutenberg on iPad using the latest iOS.

### Screenshots, screen recording, code snippet

![image](https://github.com/WordPress/gutenberg/assets/13466873/64f3358a-99b3-4830-a90f-b663758c66c5)

### Environment info

WordPress 6.3 running Twenty Twenty theme.
Google Chrome browser / Safari.
IOS.

Works *fine* with Chrome in Windows 10.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

No

**Comment by t-hamano** [comment]
Thanks for the report.

I actually tested it on my iPad as well, but could not reproduce it. The block toolbar is visible and operable whether the "top toolbar" is enabled or disabled. Is there something I am missing?

- Use InstaWP
- WordPress Version: 6.3
- iPad: OS 16.6
- Theme: Twenty Twenty
- Broser: Chrome

https://github.com/WordPress/gutenberg/assets/54422211/d6ae3eaf-bf77-44f5-8bf0-38df492905a5

**Comment by ColinRobbins** [comment]
Thanks for that.

I guess it could be a plugin causing my issue, but not sure why any should "tamper" with that.
It was working fine with the previous WordPress.

My challenge is I only have one small live system, with no test / pre-prod capability so can't go disabling plugins to rule them out.

**Comment by meNeither** [comment]
Same for Safari Version 16.2 (18614.3.7.1.5), macOS Ventura 13.1 (22C65).

If you remove `overflow: hidden;` from `.interface-interface-skeleton__content`, the block toolbar becomes visible again.

**Comment by ColinRobbins** [comment]
> If you remove `overflow: hidden;` from `.interface-interface-skeleton__content`, the block toolbar becomes visible again.

Thanks for taking the time to reply.
 Sorry, relative newbie to WordPress - where do I find the `.interface-interface-skeleton__content` file?

**Comment by meNeither** [comment]
> Thanks for taking the time to reply. Sorry, relative newbie to WordPress - where do I find the `.interface-interface-skeleton__content` file?

I apologize for my misleading comment. This was meant as a hint for the bugfixing of the Gutenberg editor.

**Comment by t-hamano** [comment]
@meNeither 

So that would mean that this problem occurs not only on the iPad, but also on the desktop?  I would be glad to know if you check the "Top toolbar" and if that makes a difference in the results.

**Comment by meNeither** [comment]
@t-hamano Yes, this also happens under macOS and yes, "Top Toolbar" is activated.

I just noticed, however, that it only happens when I add a "[Visual Link Preview](https://wordpress.org/plugins/visual-link-preview/)" block to the post.

I guess it's not a core problem after all‽

**Comment by ColinRobbins** [comment]
> I guess it's not a core problem after all‽

Does the issue occur with previous WP versions?   If so, then I’d argue it is a core issue, on the basis some change in core has caused the issue to show itself.

**Comment by meNeither** [comment]
> > I guess it's not a core problem after all‽
> 
> Does the issue occur with previous WP versions? If so, then I’d argue it is a core issue, on the basis some change in core has caused the issue to show itself.

You might be right, because it also happens when you embed a tweet, as I just discovered.

**Comment by ColinRobbins** [comment]
So, if I create a new post, the toolbar is there.

However, if I embed a tweet, it disappears.
I guess my original issue was one of the blocks affecting it.

![image](https://github.com/WordPress/gutenberg/assets/13466873/1218ace9-9755-4d00-8513-d64b89a3a6d7)

![image](https://github.com/WordPress/gutenberg/assets/13466873/e5f5799b-f6a9-49e4-b9d6-76ca5cbca2ed)

**Comment by t-hamano** [comment]
I could not reproduce it in my environment. Maybe it is because my computer is a windows OS.

@draganescu Do you know anything about this issue? Because you seem to be working on improving the block toolbar lately 🙇

https://github.com/WordPress/gutenberg/assets/54422211/a2fbddbf-c8b4-4f6c-ac46-671e12e8dc62

**Comment by ColinRobbins** [comment]
It works fine in Windows. 
The issue is on iOS?

**Comment by meNeither** [comment]
To clarify, it is a problem that occurs in Safari (iOS, iPadOS and macOS).

It has something to do with the `overflow` property of the parent elements of `.block-editor-block-toolbar`.

I have not checked the exact cause, but if, for example, `.interface-interface-skeleton__body` is set to `overflow: visible;` (instead of `overflow: auto;`), it works as desired.

The above example is not a final solution, as I do not know if the changes lead to regressions. It is only meant to serve as a hint.

**Comment by ddryo** [comment]
I am also encountering the same problem.
I have tried everything and the conditions under which it probably occurs are as follows.

Preconditions: can occur when **custom fields** are added below the post editor.

1. the problem always occurs if the length of the posted content exceeds the height of the browser.
2. even if the post content is short, this problem occurs when the custom field setting field is opened.

Also, it seems to occur only in Safari browser. (It does not occur in Chrome.) 

### Screenshot

https://github.com/WordPress/gutenberg/assets/31400297/53124c0f-0b78-405f-bb57-2b63689dacd3

This video is based on the TT3 theme and includes Yoast SEO as an example of a plugin that adds custom fields.

**Comment by draganescu** [comment]
Howdy folks! So far I've been able to repro one bug - in the 6.3 release on ios the top toolbar scrolls with content - thus becomes invisible. 

With the Gutenberg plugin active it sticks to the top - either a missed cherry pick or some new bug that only happens in core. 👀

**Comment by draganescu** [comment]
@ddryo I can repro the safari bug. working on figuring out what's going on

**Comment by andreaslindahl** [comment]
I can verify that this bug occurs in Safari 16.5.2 on macOS Ventura 13.4.1 (c)

**Comment by draganescu** [comment]
I fiddled a bit with this and the containers that could hide the top toolbar seem unchanged in terms of style props between custom fields visible and custom fields invisible. Same overflow settings, same display settings.

**Comment by draganescu** [comment]
One difference I noticed is that when there are metaboxes the visual editor container scrolls on top of the header:


**Metaboxes:**

<img width="1433" alt="Screenshot 2023-08-14 at 11 49 12" src="https://github.com/WordPress/gutenberg/assets/107534/459104b2-114e-42a3-9e20-5a7417f7008e">


**No metaboxes:**

<img width="1433" alt="Screenshot 2023-08-14 at 11 49 36" src="https://github.com/WordPress/gutenberg/assets/107534/d5eb3f94-2ca0-4b49-ab08-78aa10a9c6a5">

**Comment by t-hamano** [comment]
I feel like this issue is similar to #38893 🤔

**Comment by t-hamano** [comment]
My guess is that `z-index` may not work as expected in Safari under certain conditions.

For example, the following problems have been reported:
[Z-Index not working on Safari - fine on Firefox and Chrome - Stack Overflow](https://stackoverflow.com/questions/40895387/z-index-not-working-on-safari-fine-on-firefox-and-chrome)

**Comment by ColinRobbins** [comment]
Just to be clear, the issue I originally reported at the top of this thread occurs on both Safari and Chrome under IOS.   My screenshots above are from Chrome.

**Comment by t-hamano** [comment]
> Just to be clear, the issue I originally reported at the top of this thread occurs on both Safari and Chrome under IOS. My screenshots above are from Chrome.

Thanks for the info. However, [this comment](https://github.com/WordPress/gutenberg/issues/53495#issuecomment-1676366865) mentions that it works correctly in Chrome. So perhaps we should reiterate what OS, what device, and what browser this problem occurs on.

**Comment by meNeither** [comment]
> Thanks for the info. However, [this comment](https://github.com/WordPress/gutenberg/issues/53495#issuecomment-1676366865) mentions that it works correctly in Chrome. So perhaps we should reiterate what OS, what device, and what browser this problem occurs on.

Every browser on iOS must use the same render engine as Safari: WebKit.

That means, unless I'm completely wrong, it affects desktop Safari and all browsers on iOS.

**Comment by meNeither** [comment]
I am just realising that the behaviour might have nothing to do with Embeds or Custom Fields, but with the height of the editor and the viewport of the browser.

https://github.com/WordPress/gutenberg/assets/868080/bf4fd77b-63f3-4d71-bbdd-da6803b87e24

**Comment by meNeither** [comment]
This change fixes the bug. But I don't know if it leads to regressions in other browsers or setups. That would have to be tested. :)
![gutenberg-issue-53495-fix](https://github.com/WordPress/gutenberg/assets/868080/1c95580e-ae57-421e-b84c-808e85d79e5c)

**Comment by ddryo** [comment]
@meNeither
Thanks for the modified code suggestion.
When I added this code, the toolbar was displayed.

However, it seems that the block sidebar no longer sticks to the right edge of the screen and disappears when scrolling content.

To prevent this new problem, along with setting `.interface-interface-skeleton__conten` to `overflow: visible`, we may also need to additionally modify CSS as follows:


```diff
@media (min-width: 782px)
.interface-interface-skeleton__secondary-sidebar, .interface-interface-skeleton__sidebar {
-   position: relative!important;
+   position: sticky!important;
    width: auto;
}

**Comment by jasmussen** [comment]
I have managed to reproduce on core running an ipad in a simulator with no plugins added of note. 

<details><summary>Previous investigation</summary>

It appears that it doesn't always happen, but can happen both in fullscreen and not fullscreen modes:

<img width="1372" alt="Screenshot 2023-08-15 at 08 57 31" src="https://github.com/WordPress/gutenberg/assets/1204802/8fdce3e4-65df-4ac2-a0e3-a3b884ad2b5d">

<img width="1372" alt="Screenshot 2023-08-15 at 09 01 11" src="https://github.com/WordPress/gutenberg/assets/1204802/b7570be1-d581-4d53-adb8-c8e3bad2a66e">

I'm currently debugging to try and find out what the main difference is, but it isn't clear that this is a new regression since the structure that positions the main action toolbar with publish and save hasn't changed its behavior. I'll return when I know more.

</details>

Edit: It looks like the above is an unrelated issue.

**Comment by jasmussen** [comment]
I've actually reproduced this time, and I have a proof of concept fix in #53665. There's a bit of a risk to that PR, though, as the overflow property removed was added there for a reason, so I would appreciate thorough testing.

**Comment by draganescu** [comment]
Looks like it may be a `z-index` issue specific to Safari - see #53688
